dcos/marathon.py

```diff
@@ -54,6 +54,11 @@ def _to_exception(response):
     if isinstance(response, Exception):
         return DCOSException(_default_marathon_error(str(response)))
 
+    if response.status_code == 400:
```

Comment:
We only know the `message` field to be in a 404 response, right?  So this should be `!= 404`